doc/beamline_completion.md

```diff
@@ -71,25 +71,84 @@ This is a breaking change so would have to happen at some controlled point.
+
+The net result of these differences is that the activities and timescales differ from those presented in the above section. We refer to these as activities rather than phases, since the order of the activities is not known and possibly not interdependent.
+
+### Activity 1 - Functional wrappers to Dataset for Geometry
+
+Need to provide classes of the form `SpectrumInfo`, `ComponentInfo` and `DetectorInfo` that wrap `Dataset`. These classes should point to a `Dataset` containing all required information. Note that `ComponentInfo` and `DetectorInfo` are inter-dependent, a higher-level grouping may be required. `SpectrumInfo` uses `DetectorInfo`. We would need to consider time spent to reproduce functional and performance testing.
+
+### Activity 2 - Saving Loading
+
+As described above, no need to preserve compatibility with existing intermediate formats as far as the total `Dataset` is concerned. Time would need to be invested to trial and optimise these. Requirements for end-users should also be considered, for example if `Dataset` libraries are used outside of Mantid. Note that `Saving` and `Loading` and format specifics must not be mixed into the `Dataset` library. 
+
+### Activity 3 - Serialization/De-serialization NexusGeometry
+
+I have split this from activity above for estimation purposes.
+
+We will also need to serialize/de-serialize between in-memory geometry and target formats. For example serializing `ComponentInfo` to `NexusGeometryFormat` would likely be required for some kind of `SaveBeamline` (a rare example of something useful between Mantid as of v3.14 and with dataset).
+
+### Activity 4 - Physical Instrument
+
+This is a relatively minor consideration. Favoured approaches are to either lazy load the physical instrument as part of the instrument visualisation or to attach it as some extra variables. Infact, only `positions` and `rotations` should be allowed to differ between neutronic and physical beamlines.
+
+### Activity 5 - Advise on Parameter Map
+
+Probably should research and provide best practice for storing ParameterMap equivlent information
+
+### Notes
+
+This is slightly beyond the remit of the discussion here, but one concern with the `Dataset` rollout plan is the surge in technical debt. Conceptional understanding for a time will be problematic as there will be two very different ways of supporting data reduction. The size of the codebase will also increase significantly for a time. It may be many years before the codebase can be collapsed. 
+
+### Dataset Geometry Estimates
+
+Effort based on above consideration. **Note when comparing with Workspace 1.0 estimates that this does not include effort to take away existing approaches**
+
+| Activity  | Lower Estimate (Weeks) | Upper Estimate (Weeks) |  Notes |
+| ------------- | ------------- | ------------- |------------- |
+| 1  | 2  | 8 | Mantid already provides a good template for this. Work may be rapid |
```

Comment:
Yes should be straight forward, we are only changing data access. I think the lower estimate could be decreased, but i'm also factoring in time for performance issues and doing something about that. Tests would probably need significant rewriting to setup. If it's just `ComponentInfo` that's needed, then fine, but if we also want to extend this to cover aspects of `DetectorInfo` (you mention this above) we would need to think about redesign.

Reply:
Since nothing of the internals would change I do not see how there could be performance issues --- unless it is something inherent in the `Dataset` design (in that case we can either do nothing about it, or it should be taken into account already by items for optimization I have on the generic list).

Reply:
Good point, that would be a larger problem with dataset.